scripts/macos/RenameMacUserNameAndHomeDirectory.sh

```diff
@@ -1,61 +1,137 @@
-#!/bin/bash
-
-#
-# Script to rename the user name of a user on OS X
-#
-# The script updates the users record name and home directory
-# name from an old name to a new one.
-#
-# NOTE: MUST BE RUN AS ROOT!
-#
+#!/bin/bash -eu
+
+: '
+ Script to rename the username of a user account on OS X
+
+ The script updates the users record name, RealName (displayName), and home directory
+
+ If the user recieving the name change is signed in they will be signed out. 
+
+ Example usage: 
+
+ sudo sh RenameMacUserNameAndHomeDirectory.sh cat dog
+
+ Above example would rename account cat to dog
+
+ NOTE: SCRIPT MUST BE RUN AS ROOT!
+'
+
 abort() {
 	errString=${*}
 	echo "$errString"
 	exit 1
 }
 
-if [[ ${#} -ne 2 ]]
-then
+if [[ ${#} -ne 2 ]]; then
 	echo "Usage: $0 oldUserName newUserName"
 	exit 1
 fi
 
 oldUser=$1
 newUser=$2
 
-if [[ -z "${newUser}" ]]
-then
-	abort "New user name must not be empty!"
+#Verify valid username
+if [[ -z "${newUser}" ]]; then
+	echo "New user name must not be empty!"
+	exit 1
+fi
+
+#Test to ensure account update is needed
+if [ "$oldUser" = "$newUser" ]; then
```

Comment:
There are a few instances of bracket inconsistency, `[ condition ]` vs. `[[ condition ]]`. It would be good to pick one and stick with it. The newer syntax, and I believe currently accepted standard is the `[[ condition ]]` version. 
The major difference is that with double brackets quotes are not strictly required, spaces in values get properly evaluated. But I believe we (JC) have been keeping quotes.